src/viam/app/app_client.py

```diff
@@ -0,0 +1,791 @@
+        return FragmentPB(
+            id=self.id,
+            name=self.name,
+            fragment=dict_to_struct(self.fragment) if self.fragment else None,
+            organization_owner=self.organization_owner,
+            public=self.public,
+            created_on=datetime_to_timestamp(self.created_on) if self.created_on else None,
+            organization_name=self.organization_name,
+            robot_part_count=self.robot_part_count,
+            organization_count=self.organization_count,
+            only_used_by_owner=self.only_used_by_owner,
+        )
+
+
+class AppClient:
+    """gRPC client for method calls to app.
+
+    Constructor is used by `ViamClient` to instantiate relevant service stub. Calls to `AppClient`  methods should be made through
+    `ViamClient`.
+    """
+
+    def __init__(self, channel: Channel, metadata: Mapping[str, str], location_id: Optional[str] = None):
+        """Create an `AppClient` that maintains a connection to app.
+
+        Args:
+            channel (grpclib.client.Channel): connection to app.
+            metadata (Mapping[str, str]): Required authorization token to send requests to app.
+            location_id (Optional[str]): Default location ID.
+        """
+        self._metadata = metadata
+        self._location_client = AppServiceStub(channel)
+        self._location_id = location_id
+
+    _location_client: AppServiceStub
+    _metadata: Mapping[str, str]
+    _location_id: str
+
+    async def get_user_id_by_email(self):
+        raise NotImplementedError()
+
+    async def create_organization(self):
+        raise NotImplementedError()
+
+    async def list_organizations(self):
+        request = ListOrganizationsRequest()
+        response: ListOrganizationsResponse = await self._location_client.ListOrganizations(request, metadata=self._metadata)
+        return response.organizations
+
+    async def list_organizations_by_user(self):
+        raise NotImplementedError()
+
+    async def get_organization(self):
+        raise NotImplementedError()
+
+    async def get_organization_namespace_availability(self):
+        raise NotImplementedError()
+
+    async def update_organization(self):
+        raise NotImplementedError()
+
+    async def delete_organization(self):
+        raise NotImplementedError()
+
+    async def list_organization_members(self):
+        raise NotImplementedError()
+
+    async def create_organization_invite(self):
+        raise NotImplementedError()
+
+    async def update_organization_invite_authorizations(self):
+        raise NotImplementedError()
+
+    async def delete_organization_member(self):
+        raise NotImplementedError()
+
+    async def delete_organization_invite(self):
+        raise NotImplementedError()
+
+    async def resend_organization_invite(self):
+        raise NotImplementedError()
+
+    async def create_location(self, organization_id: str, name: str, parent_location_id: Optional[str] = None) -> Location:
```

Comment:
This might be extremely opinionated, but I feel like the ordering of these args should be different. I think it should be `name` first, then `org` and `parent`. This new ordering is about the new location, and then where that new location lives. The current argument order is where the location lives, then the name, and then where the location lives again.